### PR TITLE
Migrate to the pasta_curves crate

### PR DESCRIPTION
**Pull request by str4d**
This enables us to decouple further improvements to the Pasta implementation (e.g. GPU backend) from development of the `halo2` crate, and also enables using `pasta_curves` in places where we don't need a `halo2` dependency, such as RedPallas.

### CODE REVIEW

**Review by daira — APPROVED** [review]
utACK

**Comment by str4d** [comment]
The corresponding PR to create `pasta_curves` is https://github.com/zcash/pasta_curves/pull/3

File: Cargo.toml [hunk 1]
```diff
@@ -40,16 +40,14 @@ harness = false
 
 [dependencies]
 backtrace = { version = "0.3", optional = true }
-subtle = "2.3"
 crossbeam-utils = "0.8"
 ff = "0.9"
 group = "0.9"
 metrics = "0.14.2"
 num_cpus = "1.13"
+pasta_curves = { git = "https://github.com/zcash/pasta_curves.git", branch = "main" }
```
Comment (ebfull) [inline Cargo.toml:48]: Can this be specific to the revision currently on `main`? We don't want changes to `pasta_curves`'s git repo to break halo2 code suddenly.

**Review by ebfull — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (str4d) [reply Cargo.toml:48]: Yep, I was waiting for the other PR to merge. I'll update this after https://github.com/zcash/pasta_curves/pull/2 is merged.

**Review by ebfull — APPROVED** [review]

**Pull request merged**